### PR TITLE
Test Books Controller

### PR DESCRIPTION
**Pull request by cquiro**
## Summary

* Create spec/support/shared_contexts/authenticated_user.rb
* Include Devise::Test::ControllerHelpers in rails_helper.rb
* Write specs for BooksController#show and BooksController#index

## Trello Card

https://trello.com/c/LYQOznHR/24-crear-tests-para-rutas-index-y-show-de-book

### CODE REVIEW

File: spec/controllers/api/v1/books_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+describe Api::V1::BooksController do
+  include_context 'Authenticated User'
+
+  describe 'GET #index' do
+    let!(:books) { create_list(:book, 3) }
+
+    before do
```
Comment (lmasello) [inline spec/controllers/api/v1/books_controller_spec.rb]: You can do it in one line too :)

**Review by lmasello — APPROVED** [review]

File: app/controllers/api/v1/books_controller.rb [hunk 2]
```diff
@@ -2,11 +2,11 @@ module Api
   module V1
     class BooksController < ApiController
       def index
-        render_paginated books, each_serializer: Api::V1::BookSerializer, status: :ok
```
Comment (facucompe) [inline app/controllers/api/v1/books_controller.rb]: Hahah, you correct it here. Nice!

File: spec/controllers/api/v1/books_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      get :index
+    end
+
+    it { expect(response).to have_http_status(:ok) }
+
+    it 'responds with the books list in json format' do
+      expected = ActiveModel::Serializer::CollectionSerializer.new(
+        books, each_serializer: BookSerializer, root: false
+      ).to_json
+      expect(response.body.to_json) =~ JSON.parse(expected)
```
Comment (facucompe) [inline spec/controllers/api/v1/books_controller_spec.rb]: In the example the test says `response_body`. We usually have this helper for tests to parse the response body.
```
module Response
  module JSONParser
    def response_body
      ActiveSupport::JSON.decode(response.body) if response.present? && response.body.present?
    end
  end
end
```
Add it in **spec/support**

Comment (facucompe) [reply spec/controllers/api/v1/books_controller_spec.rb]: Also add  `config.include Response::JSONParser, type: :controller` to your rails_helper

**Review by facucompe — CHANGES_REQUESTED** [review]
God Job!

File: spec/controllers/api/v1/books_controller_spec.rb (hunk 3 above)
Comment (cquiro) [reply spec/controllers/api/v1/books_controller_spec.rb]: @facucompe Done!

**Review by facucompe — APPROVED** [review]

**Pull request merged**